### PR TITLE
pkg/trace/{agent,config,filters}:create validator struct for filtering by required tags

### LINKED ISSUE #7050: trace-agent: Add tags-based filtering
trace-agent only offers trace [filtering by Resource](https://github.com/DataDog/datadog-agent/blob/0b2da02e66907776a252e6e7a635d5652b1ab30a/pkg/trace/agent/agent.go#L180). Implement filtering by tags.

One use-case is enforcing trace tagging to enable attribution of Datadog APM usage to business units. We could do that by discarding all traces that do not have the appropriate tags.

### PR DESCRIPTION
**Pull request by nicoledanuwidjaja**
Opened PR for wrong branch. This should be correct. [Previous PR](https://github.com/DataDog/datadog-agent/pull/7315)

Based on discussion, transitioning back to a `TagValidator` struct, expanding on initial implementation.

Created new env config type for `filter_tags`:
```
 filter_tags:
    require: "<LIST_OF_KEY_VALUE_STRINGS>"
    reject: "<LIST_OF_KEY_VALUE_STRINGS>"
```
The config transforms the lists of strings of required and rejected tag key/values and stores them in a `TagRules` struct. `TagValidator` uses `TagRules` to check whether the tags in the root span's `Meta` are valid.

Fixes #7050.

### CODE REVIEW

File: pkg/trace/agent/agent_test.go [hunk 1]
```diff
@@ -404,6 +404,119 @@ func TestProcess(t *testing.T) {
 			})
 		}
 	})
+
+	t.Run("filtering", func(t *testing.T) {
```
Comment (nicoledanuwidjaja) [inline pkg/trace/agent/agent_test.go:408]: Will update tests with `[]string` tag types

File: pkg/config/config_template.yaml [hunk 2]
```diff
@@ -778,6 +778,19 @@ api_key:
...
+  ## Each rules can contain:
+  ##  * require - list of key:value strings - The list of tags, with their key and value, that are required for a valid trace.
+  ##  * reject - list of key:value strings - The list of tags, with their key and value, that are rejected for a valid trace.
+  ##
+  ## See https://docs.datadoghq.com/tracing/guide/security/#replace-rules
+  ##
+  #
+  # filter_tags:
+  #   - require: "<LIST_OF_KEY_VALUE_STRINGS>"
+  #     reject: "<LIST_OF_KEY_VALUE_STRINGS>"
```
Comment (sarina-dd) [inline pkg/config/config_template.yaml:792]:
```suggestion
  #     - reject: "<LIST_OF_KEY_VALUE_STRINGS>"
```

**Review by sarina-dd — COMMENTED** [review]
Docs review completed

**Comment by nicoledanuwidjaja** [comment]
Addressed by https://github.com/DataDog/datadog-agent/pull/7386.

**Pull request closed without merging**